### ISSUE TITLE
Cloudlfare workers tcp connector

### ISSUE THREAD

**Issue by theetherGit** [redis/ioredis#1814]
As Cloudflare has added support for tcp sockets, we can have a cloudflare redis connector.

**Comment by only-cliches** [comment]
The library has other NodeJS dependencies like `tis` and `net`, you'd have to swap those out as well.  Not as easy as just writing a new connector.

https://github.com/redis/ioredis/blob/main/lib/types.ts

**Comment by theetherGit** [comment]
I agree, I love the way postgreg.js and node-postgres does this. But they had support from cloudflare team too.

**Comment by aleclarson** [comment]
Found this: https://github.com/kane50613/redis-on-workers

**Comment by akazwz** [comment]
Cloudflare Workers can connect to Redis using TCP with the Node.js compatibility layer enabled. This is the test repo: https://github.com/akazwz/workers-redis

**Comment by jokester** [comment]
> Cloudflare Workers can connect to Redis using TCP with the Node.js compatibility layer enabled. This is the test repo: https://github.com/akazwz/workers-redis

This looks great! @akazwz Can I ask how should the Redis instance be configured? I imaging a TLS enabled plain Redis instance on internet?

**Comment by akazwz** [comment]
@jokester It is TLS enabled, I am use the upstash redis for test

**Comment by hhquitiaqueze** [comment]
@akazwz looks super great and potential. Can I ask if it support all apis of ioredius ? (specifically the pubsub api)

**Comment by akazwz** [comment]
> [@akazwz](https://github.com/akazwz) looks super great and potential. Can I ask if it support all apis of ioredius ? (specifically the pubsub api)

no, workers does not have state, but maybe durable object support

**Comment by diadal** [comment]
> > [@akazwz](https://github.com/akazwz) looks super great and potential. Can I ask if it support all apis of ioredius ? (specifically the pubsub api)
> 
> no, workers does not have state, but maybe durable object support

I ran into this same problem and ended up building a solution — ZedGi (zedgi.app). Instead of trying to make ioredis work natively in Workers by swapping out ‎`tls` and ‎`net`, ZedGi takes a different approach: it sits in front of your Redis instance as a TCP-to-HTTP proxy, so you interact with Redis through a simple HTTPS endpoint using any HTTP client.

This means no driver patching, no dependency on ‎`cloudflare:sockets`, and it works in Workers, Containers, or any serverless runtime. Connection pooling and encryption are handled at the proxy layer.

You can try it here: https://zedgi.app 

Docs: https://docs.zedgi.app

Feel free to contact if you need help

**Comment by PavelPashov** [comment]
Thanks for the suggestion. Cloudflare Workers now supports [`node:net`](https://developers.cloudflare.com/workers/runtime-apis/nodejs/net/) and [`node:tls`](https://developers.cloudflare.com/workers/runtime-apis/nodejs/tls/) through `nodejs_compat`, so the preferred approach is to use standard ioredis with Workers’ Node.js compatibility mode rather than add a Cloudflare-specific connector to ioredis core.

If ioredis fails with the current `nodejs_compat` implementation, please open a focused issue with a minimal reproduction and the Worker compatibility date and flags.

**Comment by diadal** [comment]
> Thanks for the suggestion. Cloudflare Workers now supports [`node:net`](https://developers.cloudflare.com/workers/runtime-apis/nodejs/net/) and [`node:tls`](https://developers.cloudflare.com/workers/runtime-apis/nodejs/tls/) through `nodejs_compat`, so the preferred approach is to use standard ioredis with Workers’ Node.js compatibility mode rather than add a Cloudflare-specific connector to ioredis core.
> 
> If ioredis fails with the current `nodejs_compat` implementation, please open a focused issue with a minimal reproduction and the Worker compatibility date and flags.

Thanks — that makes sense.

Agreed that with nodejs_compat exposing node:net / node:tls, standard ioredis is the right path rather than a Cloudflare-specific connector.

The rough edge for me is connection lifecycle, not the command API. ioredis is built around a long-lived client — reconnect, offline queue, keepalive, dedicated pub/sub connections — but Workers can’t create a socket in global scope or share it across requests. So the usual “reuse one client/pool” pattern doesn’t apply, and you end up paying connect + TLS per request with careful cleanup afterward. Pub/sub is the clearest mismatch, since it wants a connection that stays open — and even the Durable Objects route has a ceiling, since an open socket only keeps the DO alive for around 15 minutes.

Happy to file a focused issue with a minimal repro plus compatibility date and flags, if documenting the expected usage (and limits) on Workers would help.
